evap/grades/views.py

```diff
@@ -186,3 +191,14 @@ def delete_grades(request, semester_id, course_id, grade_document_id):
             grade_document=grade_document,
         )
         return render(request, "grades_delete.html", template_data)
+
+
+@staff_required
+def semester_grade_activation(request, semester_id, active):
+    active = active == 'on'
+
+    activation, created = SemesterGradeDownloadActivation.objects.update_or_create(
+        semester=Semester.objects.get(id=semester_id),
```

Comment:
please use `semester = get_object_or_404(Semester, id=semester_id)` at the beginning of the method and then use `semester` here